slsim/Halos/halos.py

```diff
@@ -0,0 +1,758 @@
+from scipy import integrate
+from colossus.lss import mass_function
+from colossus.cosmology import cosmology as colossus_cosmo
+from hmf.cosmology.growth_factor import GrowthFactor
+import numpy as np
+import warnings
+from astropy.units.quantity import Quantity
+from lenstronomy.Cosmo.lens_cosmo import LensCosmo
+from slsim.Halos.halos_lens import (
+    deg2_to_cone_angle,
+    cone_radius_angle_to_physical_area,
+)
+
+
+def colossus_halo_mass_function(m_200, cosmo, z, sigma8=0.81, ns=0.96, omega_m=None):
+    """Calculate the differential halo mass function per logarithmic mass interval at a
+    given redshift.
+
+    This function leverages the Colossus library to determine the halo mass function
+    based on a mass scale (m_200), a cosmological model (cosmo), and a redshift (z). The
+    mass function, expressed as dn/dlnM, represents the number density of halos per log
+    mass interval. Parameters sigma8 and ns can be specified to adjust the calculation.
+
+    :param m_200: Halo mass scale, in solar mass units divided by the Hubble parameter
```

Comment:
I don't understand this input. Why do you need a mass scale? Can't we have a function that gets initialized ones and then can be re-used for different masses where to evaluate dn/dlnM?

Reply:
it is the range of the mass, I will change the docstring